packages/@sanity/portable-text-editor/src/editor/plugins/createWithPortableTextMarkModel.ts

```diff
@@ -389,37 +389,40 @@ export function createWithPortableTextMarkModel(
       }
     }
   }
-  /**
-   * Normalize markDefs
-   *
-   */
-  function normalizeMarkDefs(editor: PortableTextSlateEditor) {
-    const {selection} = editor
-    if (selection) {
-      const blocks = Editor.nodes(editor, {
-        at: selection,
-        match: (n) => (n as unknown as Descendant)._type === types.block.name,
-      })
-      for (const [block, path] of blocks) {
-        if (editor.isTextBlock(block)) {
-          const newMarkDefs = (block.markDefs || []).filter((def) => {
-            return block.children.find((child) => {
-              return (
-                Text.isText(child) && Array.isArray(child.marks) && child.marks.includes(def._key)
-              )
-            })
-          })
-          if (!isEqual(newMarkDefs, block.markDefs)) {
-            debug('Removing markDef not in use')
-            Transforms.setNodes(
-              editor,
-              {
-                markDefs: newMarkDefs,
-              },
-              {at: path},
+}
+/**
+ * Normalize markDefs, removing those not in use.
+ * @internal
+ */
+export function normalizeMarkDefs(
+  editor: PortableTextSlateEditor,
+  types: PortableTextMemberSchemaTypes,
+): void {
+  const {selection} = editor
+  if (selection) {
+    const blocks = Editor.nodes(editor, {
+      at: selection,
+      match: (n) => (n as unknown as Descendant)._type === types.block.name,
```

Comment:
Bonus: The casting here (`(n as unknown as Descendant)`) is no longer needed, would be great to have it removed.

Reply:
removed! Thanks